x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/ManyShardsIT.java

```diff
@@ -256,4 +259,31 @@ public void testLimitConcurrentShards() {
             }
         }
     }
+
+    public void testCancelUnnecessaryRequests() {
+        internalCluster().ensureAtLeastNumDataNodes(3);
```

Comment:
We need to use `assumeTrue(canUsePragmas())`, otherwise, this test will fail in release builds.